container/sagemaker/proxy_client.py

```diff
@@ -0,0 +1,88 @@
+# Copyright 2019 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License"). You
+# may not use this file except in compliance with the License. A copy of
+# the License is located at
+#
+#     http://aws.amazon.com/apache2.0/
+#
+# or in the "license" file accompanying this file. This file is
+# distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF
+# ANY KIND, either express or implied. See the License for the specific
+# language governing permissions and limitations under the License.
+
+import grpc
+import logging
+
+from google.protobuf import text_format
+from tensorflow_serving.apis import model_service_pb2_grpc
+from tensorflow_serving.apis import model_management_pb2
+from tensorflow_serving.config import model_server_config_pb2
+
+# TFS_GRPC_PORT = os.environ['TFS_GRPC_PORT']
+MODEL_CONFIG_FILE = '/sagemaker/model-config.cfg'
+
+logging.basicConfig(level=logging.INFO)
+log = logging.getLogger(__name__)
+
+
+class GRPCProxyClient(object):
+    def __init__(self, port, host='0.0.0.0'):
+        self.channel = grpc.insecure_channel('{}:{}'.format(host, port))
```

Comment:
is there any alternative to grpc api for this, so we can avoid the massive dependency set it introduces?

Reply:
There's no REST api for adding models (not just versions). And we don't want to modify the model-config and restart the tensorflow model server because the downtime for currently running models will be too long. GRPC call is commonly used by applications want to achieve this purpose. Or we need to change the model_server as suggested here: https://github.com/tensorflow/serving/issues/380#issuecomment-293014875

Reply:
ok, unless we also need other changes that require a custom build of TF?

Reply:
Yes.